### PR TITLE
vim.ui.input to vim.fn.input

### LINKED ISSUE #26: Use new `vim.ui.input` instead of `vim.fn.input`
**Is your feature request related to a problem? Please describe.**
`vim.ui.input` is a [newly merged](https://github.com/neovim/neovim/pull/15959) lua function for taking user text input. It is used by default in `vim.lsp.buf.rename()`. The advantage is it is 'overridable' with a custom lua function.

**Describe the solution you'd like**
Change [this line](https://github.com/nvim-treesitter/nvim-treesitter-refactor/blob/505e58a8b04596a073b326157490706ee63c3b81/lua/nvim-treesitter-refactor/smart_rename.lua#L22) to use `vim.ui.input`. The API is not exactly the same, you must provide an `on_confirm` callback to receive the user input.

### PR DESCRIPTION
**Pull request by IndianBoy42**
Fixes #26

I tried to maintain compatibility with old versions of neovim, but I don't know how to test that myself.

### CODE REVIEW

**Review by stsewd — APPROVED** [review]
Thanks! Code looks good to me, I'll test this locally in the following days.

**Comment by stsewd** [comment]
> I tried to maintain compatibility with old versions of neovim, but I don't know how to test that myself.

you can try downloading the stable version of neovim https://github.com/neovim/neovim/releases/tag/v0.5.1, if you are on linux, downloading the appimage should be easier.

File: lua/nvim-treesitter-refactor/smart_rename.lua [hunk 1]
```diff
@@ -10,39 +10,46 @@ local api = vim.api
...
-    local lsp_range = ts_utils.node_to_lsp_range(node)
-    local text_edit = { range = lsp_range, newText = new_name }
-    table.insert(edits, text_edit)
+  local node_text = ts_utils.get_node_text(node_at_point)[1]
+  local input = { prompt = "New name: ", default = node_text or "" }
+  if not vim.ui.prompt then
+    local new_name = vim.fn.input(input.prompt, input.default)
+    complete_rename(new_name)
+  else
+    vim.ui.prompt(input, complete_rename)
```
Comment (dstanberry) [inline lua/nvim-treesitter-refactor/smart_rename.lua]: Shouldn't this be `vim.ui.input`?

Comment (IndianBoy42) [reply lua/nvim-treesitter-refactor/smart_rename.lua]: You're right, thats weird, I don't know how that happened, must've been a stray search and replace

**Comment by stsewd** [comment]
> I'll test this locally in the following days.

Well, that was longer than expected... Just tested it locally, looks great!

**Pull request merged**